src/Components/Components/src/NavigationManager.cs

```diff
@@ -85,6 +84,14 @@ protected set
         }
     }
 
+    /// <summary>
+    /// Gets or sets the state associated with the current navigation.
+    /// </summary>
+    /// <remarks>
+    /// Setting the state is allowed to support unit testing scenarios, but it will not trigger a navigation.
+    /// </remarks>
```

Comment:
That remark will be confusing for almost all users of this API, since almost all of them aren't making a subclass and hence don't get to call the setter. Also it's not only for testing - it's the core mechanism for how subclasses do update this for real.

I'd recommend removing the remark.